Add warning ConstraintVerifier not thread safe (#2151)
kiegroup/optaplanner: optaplanner-docs/src/modules/ROOT/pages/constraint-streams/constraint-streams.adoc

Constraint streams include the Constraint Verifier unit testing harness.
To use it, first add a test scoped dependency to the `optaplanner-test` JAR.
[WARNING]
====

> You CANNOT use the same ConstraintVerifier instance in multiple concurrent tests. It is NOT thread safe and WILL cause errors that do not make sense.

Reviewer: ```suggestion
`ConstraintVerifier` is not thread-safe and must not be used in concurrent tests.
```
